functorch/CMakeLists.txt

```diff
@@ -16,7 +16,11 @@ target_compile_definitions(${PROJECT_NAME} PRIVATE TORCH_API_INCLUDE_EXTENSION_H
 target_compile_options(${PROJECT_NAME} PRIVATE ${TORCH_PYTHON_COMPILE_OPTIONS})
 target_compile_options_if_supported(${PROJECT_NAME} "-Wmissing-prototypes")
 target_compile_options_if_supported(${PROJECT_NAME} "-Werror=missing-prototypes")
-target_link_libraries(${PROJECT_NAME} PRIVATE torch torch_python)
+if(BUILD_LIBTORCHLESS)
+  target_link_libraries(${PROJECT_NAME} PRIVATE ${TORCH_LIB} torch_python)
+else()
+  target_link_libraries(${PROJECT_NAME} PRIVATE torch torch_python)
```

Comment:
We can probably replace these if/else by setting TORCH_LIB in advance see comment https://github.com/pytorch/pytorch/pull/126328/files#r1611944303